### PR TITLE
feat: pre-login mobile app exploration

### PR DESCRIPTION
**Pull request by omerhabib26**
Description
This PR adds support for pre-login exploration.

- Feature is implemented behind a feature flag `preLoginExperienceEnabled`
- User will be able to navigate to Signup, SignIn, DiscoveryCourses & SearchCourses screen
- Back button is added on required screens for back navigation
- Navigation from SignIn to Signup is removed
- Screen Orientation supported

Here is the video of the implemented feature

https://github.com/openedx/openedx-app-android/assets/30689349/cc6671ac-893f-4593-816a-b1c06cb9eb8a

### CODE REVIEW

**Review by farhan-arshad-dev — CHANGES_REQUESTED** [review]

**Comment by volodymyr-chekyrta** [comment]
Agree with the feedback provided by @farhan-arshad-dev. 
It would be beneficial to address those concerns before moving forward.

File: auth/src/main/java/org/openedx/auth/presentation/signin/SignInFragment.kt
Comment (omerhabib26) [reply auth/src/main/java/org/openedx/auth/presentation/signin/SignInFragment.kt:325]: It was part of the ticket scope

File: discovery/src/main/java/org/openedx/discovery/presentation/search/CourseSearchFragment.kt
Comment (omerhabib26) [reply discovery/src/main/java/org/openedx/discovery/presentation/search/CourseSearchFragment.kt:136]: fixed

File: auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt
Comment (omerhabib26) [reply auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt:144]: Let us decide as there is no decision on how we gonna handle the backstack.

File: app/src/main/java/org/openedx/app/AppRouter.kt
Comment (omerhabib26) [reply app/src/main/java/org/openedx/app/AppRouter.kt:74]: As discussed in DC, we will proceed with current flow

File: config.yaml
Comment (farhan-arshad-dev) [reply config.yaml:50]: @volodymyr-chekyrta thoughts?

File: auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt
Comment (farhan-arshad-dev) [reply auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt:144]: As decided in the meeting current flow is expected.

File: discovery/src/main/java/org/openedx/discovery/presentation/DiscoveryFragment.kt
Comment (farhan-arshad-dev) [reply discovery/src/main/java/org/openedx/discovery/presentation/DiscoveryFragment.kt:129]: current flow is good for now.

File: discovery/src/main/java/org/openedx/discovery/presentation/search/CourseSearchFragment.kt
Comment (farhan-arshad-dev) [reply discovery/src/main/java/org/openedx/discovery/presentation/search/CourseSearchFragment.kt:18]: IMO, better to go with the default studio Formatter | Linter. Open for suggestions.

File: auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt
Comment (farhan-arshad-dev) [inline auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt:118]: batter to update with `.wrapContentWidth()`.

File: auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt
Comment (farhan-arshad-dev) [inline auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt:117]: The top space should be `90dp` in total, to match with the production.

Comment (farhan-arshad-dev) [reply auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt:117]: The bottom space should be `20dp` in total, to match with the production.

File: auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt
Comment (farhan-arshad-dev) [inline auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt:126]: The bottom space should be `40dp` in total, to match with the production.

File: auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt
Comment (farhan-arshad-dev) [inline auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt:135]: bottom space should be `10dp` in total.

File: auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt
Comment (farhan-arshad-dev) [inline auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt:195]: better to add the preview for Landscape|Tablet mode.

**Review by farhan-arshad-dev — CHANGES_REQUESTED** [review]

File: auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt
Comment (omerhabib26) [reply auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt:195]: Preview added for Tablet Mode, no need for landscape mode.

File: app/src/main/java/org/openedx/app/AppActivity.kt
Comment (farhan-arshad-dev) [reply app/src/main/java/org/openedx/app/AppActivity.kt]: pending.

File: discovery/src/main/java/org/openedx/discovery/presentation/search/CourseSearchFragment.kt
Comment (farhan-arshad-dev) [inline discovery/src/main/java/org/openedx/discovery/presentation/search/CourseSearchFragment.kt:379]: replace with `val`

File: auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt
Comment (farhan-arshad-dev) [inline auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt:53]: Previously we called this screen `Logistration`, IMO, better to update the package, and class name accordingly.

**Review by farhan-arshad-dev — CHANGES_REQUESTED** [review]
Some concerns for @volodymyr-chekyrta, and some code-related nits, other than that LGTM.

File: auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt
Comment (k1rill) [inline auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt:108]: can be replaced with:
.padding(horizontal = 16.dp, vertical = 32.dp)

File: auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt
Comment (k1rill) [inline auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt:116]: move this to import, for example 
`import org.openedx.core.R as CR`

File: discovery/src/main/java/org/openedx/discovery/presentation/DiscoveryFragment.kt
Comment (k1rill) [inline discovery/src/main/java/org/openedx/discovery/presentation/DiscoveryFragment.kt:120]: named argument here not required

**Review by k1rill — CHANGES_REQUESTED** [review]

File: auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt
Comment (omerhabib26) [reply auth/src/main/java/org/openedx/auth/presentation/preAuth/PreAuthFragment.kt:116]: updated with `coreR`

**Comment by volodymyr-chekyrta** [comment]
@k1rill @farhan-arshad-dev @HamzaIsrar12,
let's focus on this PR, waiting for your review/approval.

File: auth/src/main/java/org/openedx/auth/presentation/logistration/Logistration.kt
Comment (omerhabib26) [reply auth/src/main/java/org/openedx/auth/presentation/logistration/Logistration.kt:96]: As shown in the SignIn and SignUp screen designs, we don't need statusBarPadding here.

File: auth/src/main/java/org/openedx/auth/presentation/logistration/Logistration.kt
Comment (omerhabib26) [reply auth/src/main/java/org/openedx/auth/presentation/logistration/Logistration.kt:84]: Currently, we are not following the mentioned approach.
@volodymyr-chekyrta your thoughts on this?

File: auth/src/main/java/org/openedx/auth/presentation/signin/SignInFragment.kt
Comment (omerhabib26) [reply auth/src/main/java/org/openedx/auth/presentation/signin/SignInFragment.kt:206]: Hey @k1rill, we can do that but in my opinion, are there any benefits to taking such a long route for a single flag? is there inspiration or reference available?

Comment (k1rill) [reply auth/src/main/java/org/openedx/auth/presentation/signin/SignInFragment.kt]: @omerhabib26  we are trying to build Clean architecture with SOLID principles. UI shouldn't work with logic directly. Config can't be changed runtime at this moment, so we should init it once on the ViewModel init. Also when the new configuration will be merged, we have to update this code, it's better to update it once inside viewModel, instead of searching all entries. Much easier to make this route long while development, than to change it a lot of times while maintaining the code

File: auth/src/main/java/org/openedx/auth/presentation/logistration/Logistration.kt
Comment (k1rill) [inline auth/src/main/java/org/openedx/auth/presentation/logistration/Logistration.kt:199]: Please rename `SignInScreen`. nit: Also there are 2 different names on this screen, fragment is called `Logistration`, and main Composable is called `PreLoginScreen`

File: auth/src/main/java/org/openedx/auth/presentation/signin/SignInFragment.kt
Comment (omerhabib26) [reply auth/src/main/java/org/openedx/auth/presentation/signin/SignInFragment.kt]: updated

**Comment by omerhabib26** [comment]
@k1rill ready for another pass

File: auth/src/main/java/org/openedx/auth/presentation/logistration/Logistration.kt
Comment (k1rill) [reply auth/src/main/java/org/openedx/auth/presentation/logistration/Logistration.kt]: will leave this as nit

**Review by k1rill — DISMISSED** [review]
@omerhabib26 good job!

**Comment by omerhabib26** [comment]
@volodymyr-chekyrta the PR is approved now you can merge it. 🚀

**Comment by volodymyr-chekyrta** [comment]
@omerhabib26, I found one problem, could you please check why it is crashing?
The video is attached below.
https://github.com/openedx/openedx-app-android/assets/127732735/26e61930-5179-4912-a2ca-d318a02b2774

**Comment by volodymyr-chekyrta** [comment]
Also, there is a crash when trying to open any course.
`
java.lang.IllegalStateException: Fragment DiscoveryFragment{a096e35} (d8cc59a3-8b4b-4a42-9f65-50d120962281 id=0x7f0a00b8 tag=DiscoveryFragment) is not a child Fragment, it is directly attached to org.openedx.app.AppActivity@c8a8e38
`
https://github.com/openedx/openedx-app-android/assets/127732735/6524ad19-a10d-417a-9ba2-a11d1a045879

**Comment by omerhabib26** [comment]
> Same problem as in iOS - the user sees an error message when trying to enroll. Perhaps should redirect the user to log in or hide the "Enroll" button.
> 
> Screenshot_20231130_131802

This is a known behavior and is not a part of the ticket scope.

**Comment by omerhabib26** [comment]
> @omerhabib26, I found one problem, could you please check why it is crashing? The video is attached below. https://github.com/openedx/openedx-app-android/assets/127732735/26e61930-5179-4912-a2ca-d318a02b2774

It was intermittent but fixed now.

File: discovery/src/main/java/org/openedx/discovery/presentation/DiscoveryFragment.kt
Comment (omerhabib26) [reply discovery/src/main/java/org/openedx/discovery/presentation/DiscoveryFragment.kt]: It was not the part of the ticket scope but I have fixed it

**Comment by omerhabib26** [comment]
@volodymyr-chekyrta PR is ready for final pass

**Review by farhan-arshad-dev — CHANGES_REQUESTED** [review]
Just an improvement, other than that looks good to me.

**Review by farhan-arshad-dev — APPROVED** [review]

**Review by volodymyr-chekyrta — APPROVED** [review]

File: auth/src/main/java/org/openedx/auth/presentation/logistration/Logistration.kt
Comment (volodymyr-chekyrta) [inline auth/src/main/java/org/openedx/auth/presentation/logistration/Logistration.kt:55]: To keep consistency with other fragments, it would be great to rename it to LogistrationFragment.
Same to the file name.

**Comment by volodymyr-chekyrta** [comment]
No other issues; we can merge it after renaming the Logistration file and resolving conflicts.

**Review by volodymyr-chekyrta — APPROVED** [review]
LGTM 🚀

**Review by farhan-arshad-dev — CHANGES_REQUESTED** [review]
Some Nits
No need file `config.yaml`, can be deleted.

After that, I'll approve the PR.

**Pull request merged**
